File: Sorting_Methods/merge.py
def mergeSor(arr):
    if(len(arr))>1:
        mid=len(arr)//2
        L=arr[:mid]
        R=arr[mid:]
        mergeSor(L)
        mergeSor(R)
        
        i=j=k=0

        while i<len(L) and j < len(R):
            if L[i]<R[i]:
                arr[k]=L[i]
                i+=1
            else:
                arr[k]=R[j]
                j+=1
            k+=1

        while i<len(L):
            arr[k]=L[i]
            i+=1
            k+=1
        while j<len(R):
            arr[k]=R[j]
            j+=1
            k+=1


if __name__ == '__main__':
    arr=[5,6,1,2,3,4]
    print(arr)
    mergeSor(arr)
    print(arr)

File: sorting/merge.cpp
#include <bits/stdc++.h>
#include<iostream>
using namespace std;

void merge(int arr[],int start,int mid,,int end)
{

    for(int i=a;i<b;i++)
    {

    }
}

void mergesort(int arr[],int l,int h)
{
    if(l<h)
    {
        int m=(l+h)/2;
        mergesort(arr,0,m);
        mergesort(arr,m+1,h);
        merge(arr,l,m,h);
    }
    else
    {
        return;
    }
}
int main()
{
    int arr[]={14,7,3,12};
    int len=sizeof(arr)/sizeof(arr[0]);
    cout << len << endl;
    merge(arr,0,len);
}